### ISSUE TITLE
Use hierarchical menulist for Advanced Search field drop-down

### ISSUE THREAD

**Issue by dstillman** [zotero/zotero#145]
http://forums.zotero.org/discussion/22751/
- Creator
  - Author
  - Editor
  - Contributor
  - ...
- Item Type
- Publication
  - Blog Title
  - Book Title
  - Dictionary Title
  - Encyclopedia Title
  - Forum/Listserv Title
  - Proceedings Title
  - Program Title
  - Publication
  - Website Title

**Comment by gracile-fr** [comment]
With 9c52ebdf8b0911b722aedc32a428157db7cef35f , saved searches appear under "Collection" search condition. Generally speaking, mixing them could make the "Saved search" condition less discoverable.

Anyway, I wonder whether we could have an indicator *in the drop-down* menu to distinguish collections from saved searches (if displaying icons in the menu is too heavy, two headers would be ok).

**Comment by dstillman** [comment]
1) I suspect that change will make the searches more discoverable, not less. (That was the goal, at least.) Most people who don't know the terminology difference likely look under Collection for anything that appears in the left pane (which we call the Collections pane anyway). The plan is to put Unfiled in there as well.

2) https://github.com/zotero/zotero/commit/30f2ecd75aa053c5d6da8a1b8f6108290e6b69f5 moves most conditions under a submenu, so Collection will be much more prominent.

3) The change includes icons in the drop-down menu.

**Comment by gracile-fr** [comment]
Yes, I had seen the submenu, it's clearly an improvement. Thank you.
As for the icons, I was not clear. If you navigate into the menu with your keyboard, indeed you'll get the relevant icon before the name of the selected Collection / Saved search. But if you click to open the menu, there's no icon until you select a Collection / Saved search. What about adding two headers:  "Collections" at the top, "Saved Searches" lower ? (i.e. add a hint to distinguish them, like the minus sign, at the moment, helps to identify the subcollections)

(also, I found a small glitch: if you create a search with a "Collection is" condition, then change the Library, the menu is not updated with the relevant collections and saved searches, i.e. you need to set another condition before selecting again "collection is"…)

**Comment by dstillman** [comment]
In r54 the drop-down should refresh after a library change.

> if you click to open the menu, there's no icon until you select a Collection / Saved search

Not sure what you mean here. I haven't checked on Windows/Linux, but at least on macOS as soon as you select "Collection" it selects the first item in the menu, which includes an icon.

I don't see any point in adding headers given that there are icons and since the idea here is simply to mirror the "Collections pane" that people are used to.
